instance/models/mixins/openedx_database.py

```diff
@@ -75,6 +119,100 @@ def mongo_database_names(self):
         """
         return [self.mongo_database_name, self.forum_database_name]
 
+    @property
+    def migrate_user(self):
+        """
+        Name of migration user
+        """
+        return self._get_mysql_user_name("migrate")
+
+    @property
+    def read_only_user(self):
+        """
+        Name of read_only user
+        """
+        return self._get_mysql_user_name("read_only")
+
+    @property
+    def admin_user(self):
+        """
+        Name of admin user
+        """
+        return self._get_mysql_user_name("admin")
+
+    @property
+    def global_users(self):
+        """
+        List of MySQL users with global privileges (i.e., privileges spanning multiple databases)
+        """
+        return self.migrate_user, self.read_only_user, self.admin_user
+
+    def _get_mysql_database_name(self, suffix):
+        """
+        Build unique name for MySQL database using suffix.
+
+        To generate a unique name, this method adds an underscore and the specified suffix
+        to the database_name of this instance.
+
+        database_name can be up to 50 characters long.
+        The maximum length for the name of a MySQL database is 64 characters.
+        To ensure that the generated name does not exceed that length,
+        suffix should not consist of more than 13 characters.
+        """
+        mysql_database_name = "{0}_{1}".format(self.mysql_database_name, suffix)
+        assert len(mysql_database_name) <= 64
+        return mysql_database_name
+
+    def _get_mysql_user_name(self, suffix):
+        """
+        Build unique name for MySQL user using suffix.
+
+        To generate a unique name, this method adds an underscore and the specified suffix
+        to the mysql_user of this instance.
+
+        mysql_user is 6 characters long.
```

Comment:
You missed a line that says something like: "The maximum length of usernames in MySQL is 16 characters."